ITE/11/README.adoc

```diff
@@ -0,0 +1,291 @@
+```
+
+If instead the ITE is applied to layouts with support for the in-toto
+Attestation Framework, the resultant step schema is as follows.
+
+```yaml
+name: string
+command: list of strings
+expectedMaterials: list of artifact rules
+expectedPredicates: list of expectedStepPredicates
+```
+
+Each `expectedStepPredicate` object has the following schema.
+
+```yaml
+predicateType: string
+expectedProducts: list of artifact rules
+expectedAttributes: list of CEL expressions
+functionaries: list of authorized functionary key IDs
+threshold: int
+```
+
+Similarly, inspections in legacy layouts are updated to the following schema.
+
+```yaml
+name: string
+expectedMaterials: list of artifact rules
+expectedProducts: list of artifact rules
+expectedAttributes: list of CEL expressions
+run: list of strings
+```
+
+And inspections in layouts with support for the Attestation Framework will have
+the following schema.
+
+```yaml
+name: string
+command: list of strings
+expectedMaterials: list of artifact rules
+expectedPredicates: list of expectedInspectionPredicates
+```
+
+Each `expectedInspectionPredicate` object has the following schema.
+
+```yaml
+predicateType: string
+expectedProducts: list of artifact rules
+expectedAttributes: list of CEL expressions
+```
+
+=== Verifying Attribute Constraints
+
+An in-toto implementation supporting these checks must support the evaluation of
+CEL expressions. During the in-toto verification workflow, after applying rules
+from `expectedMaterials` and `expectedProducts`, each rule is evaluated against
+the attestation it applies to. If the rule expression evaluates to true, the
+rule passes and verification proceeds. If the expression evaluates to false,
+verification is aborted with an error indicating the failing rule.
+
+[[motivation]]
+== Motivation
+
+in-toto attestations, whether links from the original in-toto specification or
+contextual metadata from the Attestation Framework, contain various claims. The
+link metadata format has two opaque fields, `environment` and `byproducts`, that
+can hold arbitrary information. Contextual predicates also record various
+software supply chain specific attributes, though in predefined fields rather
+than in an arbitrary manner in opaque fields. In the current in-toto
+specification, the only way to verify these attributes are through inspections
+that require executing commands or scripts during verification. If invoked via
+scripts, they need to be communicated alongside layouts to the verifier and
+there are no in-toto specific guarantees as to their integrity or if they were
+issued by the authors of the layout. Further, these scripts can be complicated
+as they need to load attestations, once again verify their signatures using keys
+in the layout, before they can actually validate the attributes in question.
+
+=== Example: Verifying Result of a Test Run
+
+The in-toto Attestation Framework has a vetted predicate capturing the results
```

Comment:
optional: link to these various predicates to make this easier to follow?

Reply:
Done!